### ISSUE TITLE
How to resume from checkpoint?

### ISSUE THREAD

**Issue by ILG2021** [voicepowered-ai/VibeVoice-finetuning#14]
@jpgallegoar-vpai

**Comment by ILG2021** [comment]
resolve in my fork.

**Comment by jdack41** [comment]
@ILG2021 how did you get work? did you use option “—resume_from_checkpoint”?

**Comment by ILG2021** [comment]
> [@ILG2021](https://github.com/ILG2021) how did you get work? did you use option “—resume_from_checkpoint”?

yes, you need resume_from_checkpoint to restore optimizer, updates, learning rate and so on. But resume_from_checkpoint will also restore lora checkpoint for the whole model, which will cause error, because we only finetune parts of the whole model. So override the load_checkpoint function as a empty function for the Trainer.
What's more, you need to save ema shadow checkpoint and restore it when resume.
